pkg/server/endpoints/authorized_entryfetcher.go

```diff
@@ -35,94 +40,109 @@ type eventsBasedCache interface {
+func (a *AuthorizedEntryFetcherEvents) PruneEventsTask(ctx context.Context) error {
 	for {
 		select {
 		case <-ctx.Done():
-			a.log.Debug("Stopping event pruner")
+			a.c.log.Debug("Stopping event pruner")
 			return ctx.Err()
-		case <-a.clk.After(a.pruneEventsOlderThan / 2):
-			a.log.Debug("Pruning events")
-			if err := a.pruneEvents(ctx, a.pruneEventsOlderThan); err != nil {
-				a.log.WithError(err).Error("Failed to prune events")
+		case <-a.c.clk.After(a.c.pruneEventsOlderThan / 2):
+			a.c.log.Debug("Pruning events")
+			if err := a.pruneEvents(ctx, a.c.pruneEventsOlderThan); err != nil {
+				a.c.log.WithError(err).Error("Failed to prune events")
 			}
 		}
 	}
 }
 
-func (a *AuthorizedEntryFetcherWithEventsBasedCache) pruneEvents(ctx context.Context, olderThan time.Duration) error {
-	pruneRegistrationEntryEventsErr := a.ds.PruneRegistrationEntryEvents(ctx, olderThan)
-	pruneAttestedNodeEventsErr := a.ds.PruneAttestedNodeEvents(ctx, olderThan)
+func (a *AuthorizedEntryFetcherEvents) pruneEvents(ctx context.Context, olderThan time.Duration) error {
+	pruneRegistrationEntryEventsErr := a.c.ds.PruneRegistrationEntryEvents(ctx, olderThan)
+	pruneAttestedNodeEventsErr := a.c.ds.PruneAttestedNodeEvents(ctx, olderThan)
 
 	return errors.Join(pruneRegistrationEntryEventsErr, pruneAttestedNodeEventsErr)
 }
 
-func (a *AuthorizedEntryFetcherWithEventsBasedCache) updateCache(ctx context.Context) error {
+func (a *AuthorizedEntryFetcherEvents) updateCache(ctx context.Context) error {
 	updateRegistrationEntriesCacheErr := a.registrationEntries.updateCache(ctx)
 	updateAttestedNodesCacheErr := a.attestedNodes.updateCache(ctx)
 
 	return errors.Join(updateRegistrationEntriesCacheErr, updateAttestedNodesCacheErr)
 }
 
-func buildCache(ctx context.Context, log logrus.FieldLogger, metrics telemetry.Metrics, ds datastore.DataStore, clk clock.Clock, cacheReloadInterval, eventTimeout time.Duration) (*authorizedentries.Cache, *registrationEntries, *attestedNodes, error) {
-	cache := authorizedentries.NewCache(clk)
+func (a *AuthorizedEntryFetcherEvents) buildCache(ctx context.Context) error {
+	cache := authorizedentries.NewCache(a.c.clk)
 
-	registrationEntries, err := buildRegistrationEntriesCache(ctx, log, metrics, ds, clk, cache, pageSize, cacheReloadInterval, eventTimeout)
+	registrationEntries, err := buildRegistrationEntriesCache(ctx, a.c.log, a.c.metrics, a.c.ds, a.c.clk, cache, pageSize, a.c.cacheReloadInterval, a.c.eventTimeout)
 	if err != nil {
-		return nil, nil, nil, err
+		return err
 	}
 
-	attestedNodes, err := buildAttestedNodesCache(ctx, log, metrics, ds, clk, cache, cacheReloadInterval, eventTimeout)
+	attestedNodes, err := buildAttestedNodesCache(ctx, a.c.log, a.c.metrics, a.c.ds, a.c.clk, cache, a.c.cacheReloadInterval, a.c.eventTimeout)
 	if err != nil {
-		return nil, nil, nil, err
+		return err
 	}
 
-	return cache, registrationEntries, attestedNodes, nil
+	a.cache = cache
```

Comment:
I think we'd need to have some kind of mutex around setting these otherwise it may race with other goroutines accessing the cache.

Reply:
thanks, good catch.

Reply:
Added the locks, also tuned back the frequency to 24 hours which i think is sufficient.